### PR TITLE
Feature/okta sdk

### PR DESCRIPTION
**Pull request by chazkiker2**
# What's here (& why)

- (include link(s) to applicable Trello card(s))

# Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# Change Status

- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete / work-in-progress, PR is for discussion or feedback
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes 
- [ ] There are no merge conflicts

### CODE REVIEW

File: src/main/java/com/lambdaschool/oktafoundation/config/OktaAuthSecurityConfig.java [hunk 1]
```diff
@@ -35,25 +41,26 @@ protected void configure(HttpSecurity http)
...
+				.permitAll()
 				.antMatchers(HttpMethod.PATCH, "/courses/**", "/modules/**")
-				.hasAnyRole("ADMIN", "TEACHER")
+				.permitAll()
 				.antMatchers(HttpMethod.PUT, "/courses/**", "/modules/**", "/students/**")
-				.hasAnyRole("ADMIN", "TEACHER")
+				.permitAll()
 				.antMatchers(HttpMethod.DELETE, "/courses/**", "/modules/**", "/students/**", "/teachers/**")
-				.hasAnyRole("ADMIN", "TEACHER")
+				.permitAll()
```
Comment (BUGBOUNTYchrisg8691) [inline src/main/java/com/lambdaschool/oktafoundation/config/OktaAuthSecurityConfig.java]: I don't think you meant to upload this version of this file. If so, just let me know why and I'll approve.

File: target/classes/application.properties [hunk 2]
```diff
@@ -22,22 +28,18 @@ spring.jackson.serialization.fail-on-empty-beans=false
...
 spring.mvc.throw-exception-if-no-handler-found=true
-spring.web.resources.add-mappings=false
\ No newline at end of file
+spring.web.resources.add-mappings=false
+spring.security.oauth2.resourceserver.jwt.issuer-uri=${OKTAOAUTH2ISSUER}
+okta.oauth2.client-id=${OKTAOAUTH2CLIENTID}
+okta.oauth2.issuer=${OKTAOAUTH2ISSUER}
+okta.oauth2.groups-claim=groups
+okta.client.token=00ADYFjWdYPkIMmVZwawR5jAhQIbe-mqz3uSTDc81m
+okta.client.org-url=https://dev-41327786.okta.com
```
Comment (BUGBOUNTYchrisg8691) [inline target/classes/application.properties]: We probably don't want the above lines up there. I always forget about `target/`

**Review by BUGBOUNTYchrisg8691 — CHANGES_REQUESTED** [review]
Overall looks amazing. The functionality we get from these changes is going to make work on the front end so much easier. There's just a few lingering things from testing that need to be changed, but other than that, this is just freaking awesome.

**Review by BUGBOUNTYchrisg8691 — APPROVED** [review]
Perfect

**Review by dav1dchang — APPROVED** [review]
Great job!

**Pull request merged**